src/fides/api/ops/api/v1/endpoints/dataset_endpoints.py

```diff
@@ -193,12 +195,23 @@ def patch_dataset_configs(
             .filter_by(fides_key=dataset_pair.ctl_dataset_fides_key)
             .first()
         )
-        fetched_dataset: Dataset = Dataset.from_orm(ctl_dataset)
+        if not ctl_dataset:
+            raise HTTPException(
+                status_code=HTTP_404_NOT_FOUND,
+                detail=f"No ctl dataset with key '{dataset_pair.ctl_dataset_fides_key}'",
+            )
+
+        try:
+            fetched_dataset: Dataset = Dataset.from_orm(ctl_dataset)
+        except PydanticValidationError as e:
+            raise HTTPException(
+                status_code=HTTP_422_UNPROCESSABLE_ENTITY, detail=e.errors()
+            )
+        validate_data_categories(fetched_dataset, db)
```

Comment:
Since we have to fetch the dataset anyway to validate for saas connectors, I"m going ahead and checking that its data categories are also in the database like we do on existing dataset config endpoints. The fix is to adjust the ctl dataset though.

It's easier to add this check here than to a generic crud endpoint.

The reason this isn't part of a standard validator is that it needs access to a database session.

Reply:
my instinct would be on resource creation since that's the closest layer to the db so it'd catch the most things. but yes, the generic CRUD endpoint is nice code cleanliness wise, but makes changes like this more difficult. maybe something to discuss with the team and make a follow up ticket for?

Reply:
It's tricky though because the data categories could change later! They may be valid at the time of creation and not at the time of use!

Reply:
will make a ticket!

Reply:
https://github.com/ethyca/fides/issues/2115